app/controllers/middleware_deployment_controller.rb

```diff
@@ -17,4 +28,48 @@ def show
     show_container(@record, controller_name, display_name)
   end
 
+  def button
+    selected_operation = params[:pressed].to_sym
+    if OPERATIONS.key?(selected_operation)
```

Comment:
Need not be a helper, you can just create a class under the controller.

Also consider using `Struct` rather that `Hash`. It's not very obvious what all the hash valid keys are compared to members in a struct.